### ISSUE TITLE
Outage of https://logs.cloud.gov

### ISSUE THREAD

**Issue by jezhumble** [18F/cg-postmortems#6]
At 7:12pm Eastern Time on 5 July, we discovered a login redirect loop on
the `https://logs.cloud.gov/` service. This issue prevented users from logging
into the service. Upon further inspection we discovered that the session
management system for the service was failing to properly save the user's login
session which caused the redirect.

After discovering that the `Redis` database was not properly saving user sessions,
we found that this was because the database was at capacity which was hard-coded
to use 1GB instead of 86% of available memory for the `Elasticsearch` cluster.

After updating and redeploying the `Elasticsearch` cluster, we discovered that
the it would not successfully restart due to hardening setting the `/tmp`
directory to `noexec`, causing `JVM` not to `JIT` jars. After successfully
restarting the instances, we discovered that the encrypting disks were further
increasing the time needed to restore the service.

While the cluster is slowly redeployed, we update the `cg-deploy-logsearch` to
use new, larger memory instances to start keeping logs while the cluster continues
to redeploy.

At 8:00pm Eastern Time on 6 July, we observed the cluster deployment is
complete. We redeploy the Kibana instance, verify the logs, and the
`https://logs.cloud.gov/` service is restored.

The following actions were proposed in order to prevent this type of outage from
happening again:
- Failure alert for “are logs coming through?”.
- Auto-scale / high-availability for `Redis` to insulate against data loss due to
  failures in more fragile components downstream.
- Extend [Incident Response Plan](https://docs.cloud.gov/ops/security-ir/) to start with non-security-specific events
- Create a [chaos bot](https://en.wikipedia.org/wiki/Chaos_Monkey) for the `https://logs.cloud.gov/` service

We will be creating storied in [our backlog](https://github.com/18F/cg-atlas/issues/) to address these,
and reviewing status in 2 months. Please add any comments you have below.

**Comment by jbarnicle** [comment]
On July 5th, 7:12pm EST we discovered a login redirect loop on  the cloud.gov log search service `https://logs.cloud.gov/` which was preventing users from logging into the service. The session management system for the log search service was failing when trying to save user session data to the `Redis` database, which resulted in the redirect loop.

Researching this issue, we found that the `Redis` database had run out of space because it was unable to push data to our `ElasticSearch` cluster.  The `ElasticSearch` cluster cluster was failing as well, due to insufficient memory limits and a configuration error that was introduced while applying hardening scripts to the ElasticSearch implementation.

With this information, we reconfigured `Redis` on the logsearch service to use a much larger memory instance to start keeping logs while the `ElasticSearch` Cluster was down.  We then updated the `ElasticSearch` configuration to correct the memory and script hardening issues and redeployed, which took some time due to a disk encryption step. On July 6, 8:00pm EST, we observed the cluster deployment was  complete.

We are in the process of creating monitoring tasks for the log system to better detect service outages when they occur.
